simulator/CMakeLists.txt

```diff
@@ -1,10 +1,10 @@
 #! cmake
-#! Building the scalar MIPS CPU simulator
+#! Build file for the scalar MIPS CPU simulator
 #! Soshin Konstantin
 #! soshinkonstantinv@gmail.com
 #! https://github.com/SoshinK
 
-cmake_minimum_required(VERSION 3.8)
+cmake_minimum_required(VERSION 3.10)
```

Comment:
Could you please explain?

Reply:
Actually, it leads to problems with Travis since it uses CMake .9.2

Reply:
`gtest_discover_tests()` provides a nice and clean way to register tests, however, it requires CMake 3.10 ([more](https://blog.kitware.com/dynamic-google-test-discovery-in-cmake-3-10/)).

As a solution, this can be added to .travis.yml to install CMake 3.10:
```yaml
before_install:
 - wget -O cmake.sh https://cmake.org/files/v3.10/cmake-3.10.0-rc1-Linux-x86_64.sh 
 - sudo sh cmake.sh --skip-license --exclude-subdir --prefix=/usr/local
```

Reply:
Seems that no. I'll try to copy CMake module with `gtest_discover_tests()` to project and include it if CMake version < 3.10